praw/helpers.py

```diff
@@ -111,6 +111,104 @@ def valid_redditors(redditors, sub):
             if resp['ok']]
 
 
+def all_submissions(reddit_session,
+                    subreddit,
+                    highest_timestamp=None,
+                    lowest_timestamp=None,
+                    verbosity=1):
+    """Yield submissions between two timestamps.
+
+    If both ``highest_timestamp`` and ``lowest_timestamp`` are unspecified,
+    yields all submissions in the ``subreddit``.
+
+    Submissions are yielded from newest to oldest(like in the "new" queue).
+
+    :param reddit_session: The reddit_session to make requests from. In all the
+        examples this is assigned to the variable ``r``.
+    :param subreddit: Either a subreddit object, or the name of a
+        subreddit. Use `all` to get the submissions stream for all submissions
+        made to reddit.
+    :param highest_timestamp: The upper bound for ``created_utc`` attribute
+        of submissions. (Default: current unix time)
+    :param lowest_timestamp: The lower bound for ``created_utc`` atributed of
+        submissions.
+        (Default: subreddit's created_utc or 0 when subreddit == "all").
+        NOTE: both highest_timestamp and lowest_timestamp are proper
+        unix timestamps(just like ``created_utc`` attributes)
+    :param verbosity: A number that controls the amount of output produced to
+        stderr. <= 0: no output; >= 1: output the total number of submissions
+        processed; >= 2: output debugging information regarding
+        the search queries. (Default: 1)
+
+    """
+    def debug(msg, level):
+        if verbosity >= level:
+            sys.stderr.write(msg + '\n')
+
+    reddit_timestamp_offset = 28800
+    if highest_timestamp is None:
+        # convert normal unix timestamps into broken reddit timestamps
+        highest_timestamp = int(time.time()) + reddit_timestamp_offset
+    else:
+        highest_timestamp += reddit_timestamp_offset
+
+    if lowest_timestamp is not None:
+        lowest_timestamp += reddit_timestamp_offset
+    elif not isinstance(subreddit, six.string_types):
```

Comment:
I actually tried this first, but I got an error. Apparently adding "from praw.objects import Subreddit" results in a circular dependency between `praw/__init__.py` and `praw/helpers.py`. I figured checking this way is simpler than trying to resolve the circular dependency. 


Here is the error. 
```
>>> import praw.helpers
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "praw/__init__.py", line 36, in <module>
    from praw.handlers import DefaultHandler
  File "praw/handlers.py", line 10, in <module>
    from praw.helpers import normalize_url
  File "praw/helpers.py", line 31, in <module>
    from praw.objects import Subreddit
  File "praw/objects.py", line 33, in <module>
    from praw import (AuthenticatedReddit as AR, ModConfigMixin as MCMix,
ImportError: cannot import name AuthenticatedReddit
>>>
```